cro-clib/src/basic_wallet.rs

```diff
@@ -0,0 +1,185 @@
+use crate::types::{CroAddress, CroAddressPtr, CroResult};
+use chain_core::init::address::RedeemAddress;
+
+use chain_core::state::account::StakedStateAddress;
+use chain_core::tx::data::address::ExtendedAddr;
+use client_common::MultiSigAddress;
+use client_common::{PrivateKey, PublicKey};
+use std::ptr;
+
+/// # Safety
+fn do_cro_basic_create_staking_address(private: PrivateKey) -> Result<Box<CroAddress>, ()> {
+    let public: PublicKey = PublicKey::from(&private);
+    let address: StakedStateAddress = StakedStateAddress::BasicRedeem(RedeemAddress::from(&public));
+    match address {
+        StakedStateAddress::BasicRedeem(redeem) => {
+            assert!(20 == redeem.0.len());
```

Comment:
this dynamic assert is redundant -- the type of `redeem.0` is `[u8; REDEEM_ADDRESS_BYTES]`.

If you want to have a piece of mind, you can have a compile-time assertion:
```
const_assert!(REDEEM_ADDRESS_BYTES, 20)
```

Reply:
got it, i'll check